### PR TITLE
Fides Cloud Configuration: DNS Record and Privacy Center URL

### PR DESCRIPTION
**Pull request by TheAndrewJackson**
Closes fidesplus#1064 Closes fidesplus#1077

### Description Of Changes

This PR updates the admin ui to use the new fides cloud config api to aid in environment configuration.

### Code Changes

* [ ] Add fides cloud config migration and model
* [ ] Add fides cloud config endpoint to plus slice and update feature slice
* [ ] Update api types ( and fix SystemHistory issue that resulted from that)
* [ ] Add DNS records page
* [ ] Update javascript tag component to use new privacy center url config var and include `Fides.gtm()`

### Steps to Confirm

* [ ] See fidesplus PR for instructions: https://github.com/ethyca/fidesplus/pull/1070

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Review by allisonking — APPROVED** [review]
awesome work! I was able to get it running and populating locally 💥 

one nit is can this copy button for gtm be right aligned?
image

File: clients/admin-ui/src/features/privacy-experience/JavaScriptTag.tsx [hunk 1]
```diff
@@ -10,17 +11,39 @@ import {
...
+    undefined,
+    {
+      skip: !isFidesCloud,
+    }
+  );
+
+  const FIDES_JS_SCRIPT_TAG = useMemo(
+    () =>
+      isFidesCloud && isSuccess && fidesCloudConfig?.privacy_center_url
+        ? FIDES_JS_SCRIPT_TEMPLATE.replace(
```
[the remark this answers is not in the document]
Comment (TheAndrewJackson) [reply clients/admin-ui/src/features/privacy-experience/JavaScriptTag.tsx:40]: thanks!

File: clients/admin-ui/src/pages/management/dns-records.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  return (
+    <Layout title="DNS Records">
+      <Box data-testid="dns-records">
+        <Heading marginBottom={4} fontSize="2xl">
+          DNS Records
+        </Heading>
+        <Box maxWidth="600px">
+          <Text marginBottom={2} fontSize="md">
+            Set the follow record on your DNS provider to continue.
```
[the remark this answers is not in the document]
Comment (TheAndrewJackson) [reply clients/admin-ui/src/pages/management/dns-records.tsx:66]: yes 😅

File: clients/admin-ui/src/features/common/nav/v2/nav-config.ts [hunk 3]
```diff
@@ -252,6 +262,12 @@ const configureNavRoute = ({
     return undefined;
   }
 
+  // If the target route would require fides cloud in a non-fides-cloud environment,
+  // exclude it from the group.
+  if (route.requiresFidesCloud && !hasFidesCloud) {
```
[the remark this answers is not in the document]
Comment (TheAndrewJackson) [reply clients/admin-ui/src/features/common/nav/v2/nav-config.ts:267]: yes! I'll add one now

[the remark this answers is not in the document]
Comment (TheAndrewJackson) [reply clients/admin-ui/src/features/common/nav/v2/nav-config.ts:267]: tests here: https://github.com/ethyca/fides/pull/4034/commits/d3ceb5656cd3bb386e1ba9cba344639628dc05d5

**Pull request merged**